clarifai/client/model.py

```diff
@@ -235,26 +257,62 @@ def predict_by_url(self, url: str, input_type: str):
         >>> model_prediction = model.predict_by_url('url', 'image')
     """
     if input_type not in {'image', 'text', 'video', 'audio'}:
-      raise UserError('Invalid input type it should be image, text, video or audio.')
-    # TODO will be obtain proto from input class
+      raise UserError(
+          f"Got input type {input_type} but expected one of image, text, video, audio.")
+
     if input_type == "image":
-      input_proto = resources_pb2.Input(
-          data=resources_pb2.Data(image=resources_pb2.Image(url=url)))
+      input_proto = Inputs().get_input_from_url("", image_url=url)
     elif input_type == "text":
-      input_proto = resources_pb2.Input(data=resources_pb2.Data(text=resources_pb2.Text(url=url)))
+      input_proto = Inputs().get_input_from_url("", text_url=url)
     elif input_type == "video":
-      input_proto = resources_pb2.Input(
-          data=resources_pb2.Data(video=resources_pb2.Video(url=url)))
+      input_proto = Inputs().get_input_from_url("", video_url=url)
     elif input_type == "audio":
-      input_proto = resources_pb2.Input(
-          data=resources_pb2.Data(audio=resources_pb2.Audio(url=url)))
+      input_proto = Inputs().get_input_from_url("", audio_url=url)
 
-    return self.predict(inputs=[input_proto])
+    return self.predict(
+        inputs=[input_proto], inference_params=inference_params, output_config=output_config)
+
+  def _override_model_version(self, inference_params: Dict = {}, output_config: Dict = {}) -> None:
+    """Overrides the model version.
+
+    Args:
+        inference_params (dict): The inference params to override.
+        output_config (dict): The output config to override.
+          min_value (float): The minimum value of the prediction confidence to filter.
+          max_concepts (int): The maximum number of concepts to return.
+          select_concepts (list[Concept]): The concepts to select.
+          sample_ms (int): The number of milliseconds to sample.
+    """
+    if inference_params is not None:
+      params = Struct()
+      params.update(inference_params)
+
+    self.model_info.model_version.output_info.CopyFrom(
+        resources_pb2.OutputInfo(
+            output_config=resources_pb2.OutputConfig(**output_config), params=params))
+
+  def load_info(self) -> None:
```

Comment:
Minor suggestion, does `get_info` more relevant, WDYT?

Reply:
- The function doesn't return anything it just loads more model info in `kwargs` so I think `load_info` is more appropriate